### ISSUE TITLE
Errors in comparison of long values

### ISSUE THREAD

**Issue by zherebjatjew** [bwaldvogel/mongo-java-server#192]
TLDR
Some values of type long are considered as equal when they aren't.

Given
  * there is a field of type `long`
  * there are two items to save
  * the values are more than 17 digits long
  * they differ in last digit only
In such case the values are treated as equal.

```java
        long id1 = 223372036854775806L;
        long id2 = 223372036854775800L;
        // GIVEN there are no items in the collection having the given ids
        assertThat(collection.find(Filters.eq(id1)).first()).isNull();
        assertThat(collection.find(Filters.eq(id2)).first()).isNull();

        // WHEN we insert an item with id #1
        collection.insertOne(new Document("_id", id1).append("name", "item 1"));

        // THEN the collections has the item
        assertThat(collection.find(Filters.eq(id1)).first()).isNotNull();
        // AND the collection DOES NOT have an item with id #2
        // --> here the assertion fails because the item with id1 is found
        assertThat(collection.find(Filters.eq(id2)).first()).isNull();
```

**Comment by zherebjatjew** [comment]
Yes, as far as I can see it is fixed. Thank you for the quick merge!

**Comment by bwaldvogel** [comment]
I just released version `1.40.0` which contains the fix.
